src/MicrosoftAi/AbstractionMapper.cs

```diff
@@ -0,0 +1,321 @@
+				Role = ToOllamaSharpRole(cm.Role),
+				ToolCalls = cm.Contents.OfType<FunctionCallContent>().Select(ToOllamaSharpToolCall),
+			};
+		}
+	}
+
+	/// <summary>
+	/// Converts a Microsoft.Extensions.AI.<see cref="DataContent"/> to a base64 image string.
+	/// </summary>
+	/// <param name="content">The data content to convert.</param>
+	private static string ToOllamaImage(DataContent content)
+	{
+		if (content is null || !content.ContainsData)
+			return string.Empty;
+
+		if (content.MediaType?.StartsWith("image", StringComparison.OrdinalIgnoreCase) ?? false)
+		{
+			return content.Data.ToString(); // TODO convert to base64?
+		}
+
+		return string.Empty;
+	}
+
+	/// <summary>
+	/// Converts a Microsoft.Extensions.AI.<see cref="FunctionCallContent"/> to a <see cref="Message.ToolCall"/>.
+	/// </summary>
+	/// <param name="functionCall">The function call content to convert.</param>
+	private static Message.ToolCall ToOllamaSharpToolCall(FunctionCallContent functionCall)
+	{
+		return new Message.ToolCall
+		{
+			Function = new Message.Function
+			{
+				Arguments = functionCall.Arguments,
+				Name = functionCall.Name
+			}
+		};
+	}
+
+	/// <summary>
+	/// Maps a <see cref="Microsoft.Extensions.AI.ChatRole"/> to an <see cref="OllamaSharp.Models.Chat.ChatRole"/>.
+	/// </summary>
+	/// <param name="role">The chat role to map.</param>
+	private static Models.Chat.ChatRole ToOllamaSharpRole(Microsoft.Extensions.AI.ChatRole role)
+	{
+		return role.Value switch
+		{
+			"assistant" => OllamaSharp.Models.Chat.ChatRole.Assistant,
+			"system" => OllamaSharp.Models.Chat.ChatRole.System,
+			"user" => OllamaSharp.Models.Chat.ChatRole.User,
+			"tool" => OllamaSharp.Models.Chat.ChatRole.Tool,
+			_ => new OllamaSharp.Models.Chat.ChatRole(role.Value),
+		};
+	}
+
+	/// <summary>
+	/// Maps an <see cref="OllamaSharp.Models.Chat.ChatRole"/> to a <see cref="Microsoft.Extensions.AI.ChatRole"/>.
+	/// </summary>
+	/// <param name="role">The chat role to map.</param>
+	private static Microsoft.Extensions.AI.ChatRole ToAbstractionRole(OllamaSharp.Models.Chat.ChatRole? role)
+	{
+		if (role is null)
+			return new Microsoft.Extensions.AI.ChatRole("unknown");
+
+		return role.ToString() switch
+		{
+			"assistant" => Microsoft.Extensions.AI.ChatRole.Assistant,
+			"system" => Microsoft.Extensions.AI.ChatRole.System,
+			"user" => Microsoft.Extensions.AI.ChatRole.User,
+			"tool" => Microsoft.Extensions.AI.ChatRole.Tool,
+			_ => new Microsoft.Extensions.AI.ChatRole(role.ToString()),
+		};
+	}
+
+	/// <summary>
+	/// Converts a <see cref="ChatResponseStream"/> to a <see cref="StreamingChatCompletionUpdate"/>.
+	/// </summary>
+	/// <param name="response">The response stream to convert.</param>
+	public static StreamingChatCompletionUpdate ToStreamingChatCompletionUpdate(ChatResponseStream? response)
+	{
+		return new StreamingChatCompletionUpdate // TODO
+		{
+			//AdditionalProperties
+			//AuthorName
+			//ChoiceIndex
+			//CompletionId
+			//Contents
```

Comment:
No need to set Contents because you're implicitly doing that by setting Text.

This assumes that Ollama streaming output only ever returns text, and not other types of content, which I think is true.